src/core/json_utils.rs

```diff
@@ -248,79 +245,57 @@ fn index_json_value<'a, V: Value<'a>>(
     }
 }
 
-// Tries to infer a JSON type from a string.
-pub fn convert_to_fast_value_and_get_term(
-    json_term_writer: &mut JsonTermWriter,
-    phrase: &str,
-) -> Option<Term> {
+/// Tries to infer a JSON type from a string and append it to the term.
+pub(crate) fn convert_to_fast_value_and_append(term: &Term, phrase: &str) -> Option<Term> {
```

Comment:
the prototype and/or the impl is a bit silly / misleading.

Reply:
I cloned it, because it's not performance critical and simplifies the code